test/account_api_test.rb

```diff
@@ -108,6 +108,33 @@
     end
   end
 
+  describe "#delete_user_by_subject_identifier" do
+    let(:subject_identifier) { "the-subject-identifier" }
+    let(:path) { "/api/oidc-users/#{subject_identifier}" }
+
+    it "responds with 204 OK if the user existed and has been deleted" do
+      account_api
+        .given("there is a valid user session, and an existing user")
```

Comment:
This is failing because there is no `provider_state` in account-api with this name.

One option would be to drop this `given`, as the subject identifier `user-id` [is guaranteed to exist](https://github.com/alphagov/account-api/blob/main/spec/service_consumers/pact_helper.rb#L37) and then use that instead of `the-subject-identifier`.

But it would probably be clearer to change this to

```ruby
account_api
  .given("there is a user with subject identifier '#{subject_identifier}'")
  .upon_receiving("a delete-user request for '#{subject_identifier}'")
  .with(method: :delete, path: path, headers: headers)
  .will_respond_with(status: 204)
```

And then add a `provider_state` to account-api called `there is a user with subject identifier 'subject-identifier'`

Reply:
Something like this?
https://github.com/alphagov/account-api/pull/158